mfr/extensions/jasp/html_processor.py

```diff
@@ -0,0 +1,63 @@
+
+from io import StringIO
+from html.parser import HTMLParser
+
+import base64
+
+class HTMLProcessor(HTMLParser):
+
+    # The HTMLProcessor replaces the src attribute in <image> tags with the base64 equivalent
+    # The image content comes from the zip_file (specified with set_src_source())
+    # It also strips <script> and <object> tags from the HTML (potential attack vectors)
+
+    html = StringIO()  # buffer for the processed HTML
```

Comment:
yeah, i was puzzled by the appending thing, but i assumed it was a bug at your end. i will add a clear statement.